model/task/task.go

```diff
@@ -2500,89 +2501,92 @@ func (t *Task) Insert() error {
 // into the old_tasks collection. If this is a display task, its execution tasks
 // are also archived.
 func (t *Task) Archive() error {
-	archiveTask := t.makeArchivedTask()
-	err := db.Insert(OldCollection, archiveTask)
-	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"archive_task_id": archiveTask.Id,
-			"old_task_id":     archiveTask.OldTaskId,
-			"execution":       t.Execution,
-			"display_only":    t.DisplayOnly,
-		}))
-		return errors.Wrap(err, "inserting archived task into old tasks")
-	}
-	err = UpdateOne(
-		bson.M{
-			IdKey:     t.Id,
-			StatusKey: bson.M{"$in": evergreen.TaskCompletedStatuses},
-		},
-		bson.M{
-			"$set": bson.M{
-				CanResetKey: true,
-			},
-			"$unset": bson.M{
-				AbortedKey:              "",
-				AbortInfoKey:            "",
-				OverrideDependenciesKey: "",
-			},
-			"$inc": bson.M{ExecutionKey: 1},
-		})
-	if err != nil && !adb.ResultsNotFound(err) {
-		return errors.Wrap(err, "updating task")
-	}
-
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+		err := ArchiveMany([]Task{*t})
+		return errors.Wrapf(err, "archiving display task '%s'", t.Id)
+	} else {
+		// Archiving a single task.
+		archiveTask := t.makeArchivedTask()
+		err := db.Insert(OldCollection, archiveTask)
 		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
+			grip.Error(message.WrapError(err, message.Fields{
+				"archive_task_id": archiveTask.Id,
+				"old_task_id":     archiveTask.OldTaskId,
+				"execution":       t.Execution,
+				"display_only":    t.DisplayOnly,
+			}))
+			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		err = UpdateOne(
```

Comment:
This update stage looks the same as the one at the top of ArchiveAll. So should we just always call ArchiveMany? (is there any reason to not call ArchiveMany here)? Or if that can't be done, can this part be refactored from each so there isn't duplicate code?

Reply:
The logic for this (archiving in to old collection and updateone) are functionally the same as the ArchiveAll. I can simply this function just to always call 
err := ArchiveMany([]Task{*t})
return errors.Wrapf(err, "archiving task '%s'", t.Id)

but @bynn was concerned about performance (doing a transaction for every task) since it might be more expensive just restarting any task than before. I think a refactor can look really nice and is semantic, let me try implementing one!

Reply:
Ah Bynn brings up a good point about the transaction. Refactor seems like a good inbetween then 👍

Reply:
Will refactor!